tests/modules/webp-uploads/webp-uploads-test.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Tests for webp-uploads module.
+ *
+ * @package performance-lab
+ */
+
+class WebP_Uploads_Tests extends WP_UnitTestCase {
+
+	/**
+	 * Test if webp-uploads is loaded and filter applied when system supports WebP.
+	 */
+	function test_webp_uploads_is_loaded() {
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$expect = array();
+		} else {
+			$expect = array(
+				'image/jpeg' => 'image/webp',
+			);
+		}
```

Comment:
The test here primarily tests the filter and is based on whatever the server supports - like the method name says, it primarily tests that the WebP module's filter call is loaded. This makes sense, but it also has limitations. The main limitation with the test here is that if the server doesn't support WebP, this doesn't really test anything, since the empty array is also the default filter value.

Therefore, in addition to the test here, it may be valuable to have two more tests that manually call the `webp_uploads_filter_image_editor_output_format` function, one time with WebP support and one time without WebP support. You could force that WebP is detected as supported or not supported by creating two dummy image editor classes (e.g. in `tests/testdata/`) and then using the `wp_image_editors` filter (see https://github.com/WordPress/wordpress-develop/blob/5.8.1/src/wp-includes/media.php#L3790) so that they are used. This way those tests can test the function for both variants and don't rely on the server environment.

What do you think?

Reply:
> The main limitation with the test here is that if the server doesn't support WebP, this doesn't really test anything, since the empty array is also the default filter value.

Right, hence my comment about running the tests against servers that both do and don't support WebP, that way the test verifies the filter works as expected.

>  You could force that WebP is detected as supported or not supported by creating two dummy image editor classes (e.g. in tests/testdata/) and then using the wp_image_editors filter (see WordPress/wordpress-develop@5.8.1/src/wp-includes/media.php#L3790) so that they are used. 

Ah, thats a great idea - by creating these dummy classes I can test against server support regardless of actual test server. I'll work on that.